src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -28,66 +68,653 @@ const ActorInit En_Box_InitVars = {
+}
+
+void EnBox_AppearOnRoomClear(EnBox* this, GlobalContext* globalCtx) {
+    func_800B8C50(&this->dyna.actor, globalCtx);
+    if (Flags_GetClearTemp(globalCtx, this->dyna.actor.room)) {
+        if (ActorCutscene_GetCanPlayNext(this->cutsceneIdxA)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutsceneIdxA, &this->dyna.actor);
+            Flags_SetClear(globalCtx, this->dyna.actor.room);
+            EnBox_SetupAction(this, func_80868AFC);
+            this->unk_1A0 = -30;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutsceneIdxA);
+        }
+    }
+}
+
+void func_80868AFC(EnBox* this, GlobalContext* globalCtx) {
+    if (func_800F22C4(this->cutsceneIdxA, &this->dyna.actor) != 0 || this->unk_1A0 != 0) {
+        EnBox_SetupAction(this, func_80868B74);
+        this->unk_1A0 = 0;
+        func_80867FBC(&this->unk_1F4, globalCtx, (this->movementFlags & ENBOX_MOVE_0x80) != 0);
+        Audio_PlaySfxAtPos(&this->dyna.actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80868B74(EnBox* this, GlobalContext* globalCtx) {
+    func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    if (this->unk_1A0 < 0) {
+        this->unk_1A0++;
+    } else if (this->unk_1A0 < 40) {
+        if (this->movementFlags & ENBOX_MOVE_0x80) {
+            this->dyna.actor.world.pos.y -= 1.25f;
+        } else {
+            this->dyna.actor.world.pos.y += 1.25f;
+        }
+        this->unk_1A0++;
+        if (this->cutsceneIdxA != -1 && ActorCutscene_GetCurrentIndex() == this->cutsceneIdxA) {
+            if (this->unk_1A0 == 2) {
+                func_800B724C(globalCtx, &this->dyna.actor, 4);
+            } else if (this->unk_1A0 == 22) {
+                func_800B724C(globalCtx, &this->dyna.actor, 1);
+            }
+        }
+    } else if (this->unk_1A0 < 60) {
+        this->alpha += 12;
+        this->unk_1A0++;
+        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+    } else {
+        EnBox_SetupAction(this, EnBox_WaitOpen);
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+    }
+}
+
+void EnBox_WaitOpen(EnBox* this, GlobalContext* globalCtx) {
+    s32 pad;
+    AnimationHeader* animHeader;
+    f32 frameCount;
+    f32 playbackSpeed;
+    EnBox_PlaybackSpeed playbackSpeedTable;
+    Player* player;
+    Vec3f offset;
+
+    this->alpha = 255;
+    this->movementFlags |= ENBOX_MOVE_IMMOBILE;
+    if (this->unk_1EC != 0 && (this->cutsceneIdxB < 0 || ActorCutscene_GetCurrentIndex() == this->cutsceneIdxB ||
+                               ActorCutscene_GetCurrentIndex() == -1)) {
+        if (this->unk_1EC < 0) {
+            animHeader = &gBoxChestOpenAnim;
+            playbackSpeed = 1.5f;
+        } else {
+            u8 playerForm;
+
+            playbackSpeedTable = sPlaybackSpeed;
+            playerForm = gSaveContext.save.playerForm;
+            animHeader = sBigChestAnimations[playerForm];
+            playbackSpeed = playbackSpeedTable.data[playerForm];
```

Comment:
```suggestion
            playbackSpeedTable = sPlaybackSpeed;
            playerForm = gSaveContext.save.playerForm;
            animHeader = sBigChestAnimations[((void)0, gSaveContext.save.playerForm)];
            playbackSpeed = playbackSpeedTable.data[((void)0, gSaveContext.save.playerForm)];
```
Out of curiosity, does this by chance work to remove the `playerform` temp?

Reply:
nope, can't remove the temp this way